### ISSUE TITLE
Trouble Installing

### ISSUE THREAD

**Issue by harpe300** [bbuchsbaum/neurosurf#1]
Hello,

When I try to install your neurosurf package, R errors out because it cannot install the rflann package (which I assume is installed along with neurosurf) (see below for R output; R version 3.6.1). I've tried installing rflann alone, but R cannot install that either.

Any idea on how I can successfully install your package?

Thank you,
Jeremy

> devtools::install_github("bbuchsbaum/neurosurf")
Downloading GitHub repo bbuchsbaum/neurosurf@master
These packages have more recent versions available.
Which would you like to update?

1: All                                      
2: CRAN packages only                       
3: None                                     
4: RcppEigen (0.3.3.5.0 -> 0.3.3.7.0) [CRAN]
5: xfun      (0.11      -> 0.12     ) [CRAN]
6: callr     (3.3.2     -> 3.4.0    ) [CRAN]

Enter one or more numbers, or an empty line to skip updates:
3
Downloading GitHub repo bbuchsbaum/neuroim2@master
These packages have more recent versions available.
Which would you like to update?

1: All                                      
2: CRAN packages only                       
3: None                                     
4: RcppEigen (0.3.3.5.0 -> 0.3.3.7.0) [CRAN]
5: xfun      (0.11      -> 0.12     ) [CRAN]

Enter one or more numbers, or an empty line to skip updates:
3
Downloading GitHub repo YeeJeremy/rflann@master
─  preparing ‘rflann’:/private/var/folders/zm/7kjqttgj2mxbfqdvkry804d00000gp/T/Rtmpp8uOGx/remotes10bf56d28c132/YeeJeremy-rflann-e0c06e7/DESCRIPTION’ ...
   checking DESCRIPTION meta-information ...
* installing *source* package ‘rflann’ ...
** using staged installation
** libs
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/Rcpp/include" -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/RcppArmadillo/include" -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include -fopenmp -I. -I.. -fPIC  -Wall -g -O2  -c RcppExports.cpp -o RcppExports.o
clang: error: unsupported option '-fopenmp'
make: *** [RcppExports.o] Error 1
ERROR: compilation failed for package ‘rflann’
* removing ‘/Library/Frameworks/R.framework/Versions/3.6/Resources/library/rflann’
Error: Failed to install 'neurosurf' from GitHub:
  Failed to install 'neuroim2' from GitHub:
  Failed to install 'rflann' from GitHub:
  (converted from warning) installation of package ‘/var/folders/zm/7kjqttgj2mxbfqdvkry804d00000gp/T//Rtmpp8uOGx/file10bf5ea3f19b/rflann_1.5.tar.gz’ had non-zero exit status
> install.packages('rflann')

**Comment by bbuchsbaum** [comment]
yes, at some point rflann got dropped from CRAN. We can still use it via the github resource: https://github.com/YeeJeremy/rflann

However, it must be compiled from source and that is where you're having a problem.

The key error is:

clang: error: unsupported option '-fopenmp'

Your compiler does not support "openmp". I refer you to this blog for a workaround.

https://thecoatlessprofessor.com/programming/cpp/r-compiler-tools-for-rcpp-on-macos/

**Comment by harpe300** [comment]
Fantastic, this worked. Thank you!

Jeremy
